internal/pkg/utils/etcdlogger/assert.go

```diff
@@ -27,3 +34,22 @@ func Assert(t assert.TestingT, expected, actual string) bool {
 
 	return wildcards.Assert(t, expected, actual)
 }
+
+// AssertFromFile compares logs captured by the KVLogWrapper.
+// Comments "// ..." and empty lines are ignored.
+func AssertFromFile(t assert.TestingT, path, actual string) bool {
+	result := false
+	data, err := os.ReadFile(path) //nolint:forbidigo // no virtual FS
+	if assert.NoError(t, err) || errors.Is(err, os.ErrNotExist) {
+		expected := string(data)
+		result = Assert(t, expected, actual)
+	}
+
+	// Dump actual state to the .out dir
+	outDir := filepath.Join(filepath.Dir(path), ".out")              //nolint:forbidigo // no virtual FS
+	filePath := filepath.Join(outDir, filepath.Base(path)+".actual") //nolint:forbidigo // no virtual FS
+	assert.NoError(t, os.MkdirAll(outDir, 0o750))                    //nolint:forbidigo // no virtual FS
+	assert.NoError(t, os.WriteFile(filePath, []byte(actual), 0o600)) //nolint:forbidigo // no virtual FS
+
+	return result
```

Comment:
```suggestion
	if err != nil && !errors.Is(err, os.ErrNotExists) {
		return
	}
	
	expected := string(data)
	result = Assert(t, expected, actual)
	// Dump actual state to the .out dir
	outDir := filepath.Join(filepath.Dir(path), ".out")              //nolint:forbidigo // no virtual FS
	filePath := filepath.Join(outDir, filepath.Base(path)+".actual") //nolint:forbidigo // no virtual FS
	assert.NoError(t, os.MkdirAll(outDir, 0o750))                    //nolint:forbidigo // no virtual FS
	assert.NoError(t, os.WriteFile(filePath, []byte(actual), 0o600)) //nolint:forbidigo // no virtual FS

	return result
```
I think this more fits into logic. So all errors except `os.ErrNotExists` are resolved as `false`. Otherwise the error occurrs or not, does not matter. Do the logic.
I expected that the `return` is inside the if statement, but it was not so I get confused.
Also you would be able to get rid of `result := false` at start of this function so the code is thinner.
Convence in Go is to catch error and return so I would assume that here the logic is the same even though it is a test :slightly_smiling_face:
If we would like always to dump the file, could be done as `defer` function.